### ISSUE TITLE
Add possibility to delete a user

### ISSUE THREAD

**Issue by klues** [asterics/AsTeRICS-Grid#189]
Currently there is no way for users to delete their account, but it should be possible.
Superlogin already provices a [method for deleting a user](https://github.com/colinskow/superlogin#superloginremoveuseruser_id-destroydbs), which should be used.

**Comment by klues** [comment]
Implemented it like this:

* on login page there is a new accordion "Advanced settings" which contains a button for deleting the currently logged-in online user: <img width="1605" height="917" alt="Image" src="https://github.com/user-attachments/assets/330d48fb-d874-4e2a-972e-e9e84da633a0" />
* I've hidden it in this accordion so that people do not use it accidentially
* the button is only active if an online user is logged-in, otherwise disabled
* clicking the button asks for confirmation: <img width="451" height="172" alt="Image" src="https://github.com/user-attachments/assets/28231b1d-a555-48b7-ba80-da6310a18379" />
* after deleting there is a success message: <img width="744" height="79" alt="Image" src="https://github.com/user-attachments/assets/c91154f8-c99d-4225-a966-6754c14ab28f" />
* for some reason deleting users which are very old and were migrated from the old [superlogin]() authentication backend cannot be deleted, see [this issue](https://github.com/perfood/couch-auth/issues/101), but for users created after May 2023 it should work.

@arasaac-dga @ms-mialingvo @willwade please test at https://grid.asterics.eu/latest/
@arasaac-dga please also update translations in crowdin.

**Comment by arasaac-dga** [comment]
​We have tested it and it works as expected. Great job!!! Some suggestions:

- **"Advanced settings" tab** perhaps is better call it as "Advanced options" due the account delete is not a setting
- When the account cannot be deleted, In the error message perhaps would be interesting to mention that accounts before may 2023 cannot be deleted. 
- **Important:** in our experience with users with autism and the problems and disasters that create to manageners when unblocking the communicators (with PIN or not) we suggest that you require the password as an additional 2F check before delete the online account. It's something easy and sure will prevent problems with accidental deletes.

**Comment by ms-mialingvo** [comment]
On Macbook firefox I get the spinner when I enter an username, on chrome "Couldn't check username". So can't test there. It's probably unrelated to this update, haven't tried to create an online user on gridbeta in a long time. I can't login to existing online users either, that has been an issue for a while.

It's too hidden for me. Given that the delete button for offline users is next to the 'open' button, I intuitively would never think to look in some fully different part of the site for the delete button for online users but I would expect it to be in the same place as for the offline users. Sure, yes, that doesn't work directly because of the logout button using that spot. I don't know how good it would look to just have it after the 'logout' button
As for safety despite visibility, what about adding a second confirmation, something like "have you downloaded a backup of the user?" "not needed / yes" "no" or so? Or an 'if you want to delete for sure, enter the word "delete" in the text field'. A password as arasaac suggest could end up with the same problem as users forgetting their password for the account.

**Comment by arasaac-dga** [comment]
> As for safety despite visibility, what about adding a second confirmation, something like "have you downloaded a backup of the user?" "not needed / yes" "no" or so? Or an 'if you want to delete for sure, enter the word "delete" in the text field'. A password as arasaac suggest could end up with the same problem as users forgetting their password for the account.

Knowing the habilities of certain users, there is no a solution because some of them write and can make math operations (that sometimes are used as confirmation). The password is the best solution and people that forget the password abandon the accounts and @klues delete the inactive users for a long time. On the other hand there is an issue opened to add the option to recover the password. If in ear future is possible to recover it, we have no problems with abandoned accounts by passwords.

> It's too hidden for me. Given that the delete button for offline users is next to the 'open' button, I intuitively would never think to look in some fully different part of the site for the delete button for online users but I would expect it to be in the same place as for the offline users. Sure, yes, that doesn't work directly because of the logout button using that spot. I don't know how good it would look to just have it after the 'logout' button

If it's unhidden as you comment, we consider that password confirmation is really mandatory and a backup suggestion or downbload is not a bad tip before to proceed with delete.

**Comment by ms-mialingvo** [comment]
> Knowing the habilities of certain users, there is no a solution because some of them write and can make math operations (that sometimes are used as confirmation). The password is the best solution and people that forget the password abandon the accounts and @klues delete the inactive users for a long time. On the other hand there is an issue opened to add the option to recover the password. If in ear future is possible to recover it, we have no problems with abandoned accounts by passwords.

You might not but others do have a problem with that. klues did this issue because of that and he said that password recovery is not possible and won't be. So, a (logged-in) user deleting itself and reopening the account with a new password is the only solution.

I think in that case the issue that actually needs to be worked on is a way to really make sure on a technical level, that the locked screen really remains locked respectively that there is a more complex way to unlock it. (While still having some way out if someone forgets the unlock code).

**Comment by arasaac-dga** [comment]
> You might not but others do have a problem with that. klues did this issue because of that and he said that password recovery is not possible and won't be. So, a (logged-in) user deleting itself and reopening the account with a new password is the only solution.

Ok, although it is not possible to recover the password (and the only solution is to abandon and create a new one), we think that the password is safer than other 2F (due in this case there is no email or phone number associated for example to send a code). Due @klues deletes the inactive accounts, we don't see problems in abandoning an account when a password is forgotten. Anyway it's our opinion but sure without a password we will have problems because many manager don't set a PIN and if @klues "unhidde" the delete option would be really unsafe.

> I think in that case the issue that actually needs to be worked on is a way to really make sure on a technical level, that the locked screen really remains locked respectively that there is a more complex way to unlock it. (While still having some way out if someone forgets the unlock code).

Yes, this must be an improvement due some users unblock the communicator with PIN with the actual trick, but we don't see that it's easy to improve it and have a backdoor to unblock it. Anyway, this is another issue not related to this.

**Comment by ms-mialingvo** [comment]
Btw, I just cancelled some plan on some site and an additional trick would be to make the "don't delete" (respectively 'don't cancel') button in the popup extra shiny, visually. Bright color. ;)

**Comment by klues** [comment]
Thanks for your feedback, I now did this:
* you have to enter the user's password in order to delete online users
* if someone cannot remember the password, but is logged in, he/she can delete the account by using the general password `delete_my_user`. I added this so that I/we can respond to requests where users want to delete an account, don't know the password, but still logged in. In this case they have the right to delete their account. If they don't know the password and aren't logged in, they have to wait for 270 days until the account is removed because of inactivity.

Additionally I tried to make the register view clearer by adding important information more clearly and removing the (in my opinion) not needed option `remember this user on this device`, which is now default `true` for every registration. Looks like this:

<img width="1015" height="735" alt="Image" src="https://github.com/user-attachments/assets/156c4550-6e31-4834-a95d-6d4289383c67" />

Please test at https://grid.asterics.eu/latest/ - should be available in about 15 minutes.
@arasaac-dga please translate the strings in crowdin.

**Comment by arasaac-dga** [comment]
Ok. Great!!! We have tested and works as expected. New strings in Crowdin transltated too.

> * if someone cannot remember the password, but is logged in, he/she can delete the account by using the general password `delete_my_user`.

Good trick!!! ;-) We understand the general password is not published and we have to comment to people that desire to delete the account and don't remember the password, isn't it?

**Comment by klues** [comment]
> Ok. Great!!! We have tested and works as expected. New strings in Crowdin transltated too.

Great, thanks! 👍 

> Good trick!!! ;-) We understand the general password is not published and we have to comment to people that desire to delete the account and don't remember the password, isn't it?

Yes, it could be published in some docs, or we only tell it to people if the request it. Probably nobody will find it in the docs 😆

**Comment by arasaac-dga** [comment]
> Yes, it could be published in some docs, or we only tell it to people if the request it. Probably nobody will find it in the docs 😆

Totally agree 😆😆😆. Yes can be mention it in documents.

**Comment by klues** [comment]
released with https://github.com/asterics/AsTeRICS-Grid/releases/tag/release-2025-08-14-14.58%2F%2B0200
